### PR TITLE
Delete old audit events according to configured storage duration

### PR DESCRIPTION
**Pull request by Martchus**
WIP because the GRU task is not triggered anywhere so far.

---

See https://progress.opensuse.org/issues/52487

### CODE REVIEW

**Comment by Martchus** [comment]
The package still builds and contains the files. I've tested on a staging instance whether it works (by running `systemctl start openqa-enqueue-audit-event-cleanup.service` manually).

File: etc/openqa/openqa.ini [hunk 1]
```diff
@@ -88,6 +88,18 @@
...
+[audit/storage_duration]
+#startup = 10
+#jobgroup = 365
+#jobtemplate = 365
+#table = 365
+#iso = 60
+#user = 60
+#asset = 30
+#needle = 30
+#other = 15
```
Comment (kalikiana) [inline etc/openqa/openqa.ini]: These values in comments suggest that they represent the defaults. However according to the documentation, and as per the code, by default nothing is cleaned up.

If this is meant as an example, I'd suggest a more verbose comment like

    # By default cleanup is disabled, see http://open.qa/docs/#_auditing_tracking_openqa_changes
    # The following categories with example values can be uncommented as needed

Comment (Martchus) [reply etc/openqa/openqa.ini]: Makes sense, I'll change it.

File: docs/Installing.asciidoc [hunk 2]
```diff
@@ -636,14 +636,38 @@ Auditing, by default enabled, can be disabled by global configuration option in
...
+needle = 30
+other = 15
+--------------------------------------------------------------------------------
+
+In this example events of the type +startup+ would be cleaned up after 10 days, events related to
+job groups after 365 days and so on. Events which do not fall into one of these categories would be
+cleaned after 15 days. By default, cleanup is disabled.
+
+Use +systemctl enable --now openqa-enqueue-audit-event-cleanup.timer+ to schedule the cleanup
+automatically every day. It is also possible to trigger the cleanup manually with by invoking
```
Comment (okurz) [inline docs/Installing.asciidoc]: "with by"?

File: etc/openqa/openqa.ini [hunk 3]
```diff
@@ -88,6 +88,20 @@
...
+# The following categories with example values can be uncommented as needed
+#startup = 10
+#jobgroup = 365
+#jobtemplate = 365
+#table = 365
+#iso = 60
+#user = 60
+#asset = 30
+#needle = 30
+#other = 15
```
Comment (okurz) [inline etc/openqa/openqa.ini]: WDYT about being explicit and adding a suffix "_days"?

**Comment by Martchus** [comment]
> to just check for any old events to delete whenever a new one is recorded?

I don't like running the cleanup code each and every time a new event is recorded. Running it only on certain events seems arbitrary.

> which is making container based employment potentially a bit harder

I don't think so. We already have multiple systemd services and this is basically just another service which can be optionally enabled. If the system or container is not running when the timer would have triggered the persistent flag should cause the timer to trigger when the system/container is started. Likely people using a "throw away" container setup don't care much about this cleanup which is likely only relevant after years of operation anyways.

File: etc/openqa/openqa.ini (hunk 3 above)
Comment (Martchus) [reply etc/openqa/openqa.ini]: The comment above already states that the durations are measured in days. I don't like a redundant suffix for each of those keys.

File: systemd/openqa-enqueue-audit-event-cleanup.service [hunk 4]
```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=Eneques an audit event cleanup task for the openQA database.
```
Comment (okurz) [inline systemd/openqa-enqueue-audit-event-cleanup.service]: s/Eneques/Enqueues/

File: systemd/openqa-enqueue-audit-event-cleanup.timer [hunk 5]
```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=Eneques an audit event cleanup task for the openQA database every day.
```
Comment (okurz) [inline systemd/openqa-enqueue-audit-event-cleanup.timer]: as above

**Pull request merged**